### ISSUE TITLE
Support for multiple json files

### ISSUE THREAD

**Issue by Kaysen98** [Tienisto/flutter-fast-i18n#43]
Hello,

my translation files are getting big very fast. I really like to have a json file for components, for services etc. split up and not in one file per language. I dont want to write my own service which merges multiple files into one to compile it with the runner afterwards.

Are you planning on introducing multiple file support per language?

Best regards

**Comment by Tienisto** [comment]
Hi,

thanks for this issue. Yes, translation files get big very fast. This is how I would do it:

```
i18n/
- component1.i18n.json
- component1_fr.i18n.json
- component1_es.i18n.json
- component2.i18n.json
- component2_fr.i18n.json
- component2_es.i18n.json
```

```dart
String a = t.component2.title;
```

Are you fine with it or do you have any ideas?

**Comment by Kaysen98** [comment]
Hi! Thanks for the fast reply :)

I really like the concept of having a folder for each language like en/de and just putting the json files in it like components.json, services.json. So the code can identify the right language files via the folder name (language tag).
I think most language libraries combine all the files for each language and prepend the file name as the root object. So for example to access a key/value pair in components.json, the path would be components.title etc. or if you create a language file for each component like appBar.json it would be appBar.title etc.

If that means you have to refactor too much, the version you provided is also fine!

Thanks in advance for looking into this :)

**Comment by Kaysen98** [comment]
i18n/
- de
- en

**Comment by Kaysen98** [comment]
```
i18n/
- de
* components.json
* services.json
- en
* components.json
* services.json
```

Like this, but I guess it would be too much of a change in your code, since the fallback language just doesnt have any language tag behind the file name e.G. strings.json (fallback) strings_en.json ...

So the version you provided is absolutely fine

**Comment by Tienisto** [comment]
You can expect this feature to land in a month. Currently, I am too busy :)

**Comment by Kaysen98** [comment]
> You can expect this feature to land in a month. Currently, I am too busy :)

Okay thanks :)

**Comment by Tienisto** [comment]
Released in `5.5.0`

**Comment by Kaysen98** [comment]
> Released in `5.5.0`

Thanks for implementing it! I really appreciate it - sorry for answering just know, just forgot
